frontend/src/concepts/pipelines/content/createRun/RunForm.tsx

```diff
@@ -75,6 +107,9 @@ const RunForm: React.FC<RunFormProps> = ({ data, onValueChange, isCloned }) => {
           data={data.nameDesc}
           setData={(nameDesc) => onValueChange('nameDesc', nameDesc)}
           maxLength={RUN_NAME_CHARACTER_LIMIT}
+          onNameBlur={onNameBlur}
+          onNameChange={() => setHasDuplicateName(false)}
+          {...(hasDuplicateName && { nameHelperText: <DuplicateNameHelperText name={name} /> })}
```

Comment:
do you need toi use the spread operator here? can you just

```tsx
nameHelperText: hasDuplicateName  ? <DuplicateNameHelperText name={name} /> : undefined

Reply:
What you suggested also works, but why would you prefer it?

Reply:
just code clarity. i would see it used if we were spreading more than one prop conditionally